planning/behavior_velocity_planner/src/scene_module/detection_area/scene.cpp

```diff
@@ -203,6 +203,67 @@ bool DetectionAreaModule::modifyPathVelocity(PathWithLaneId * path, StopReason *
   return true;
 }
 
+// calc smallest enclosing circle with average O(N) algorithm
+std::pair<lanelet::BasicPoint2d, double> calcSmallestEnclosingCircle(
+  const lanelet::ConstPolygon2d & poly)
+{
+  const double eps = 1e-5;
+  lanelet::BasicPoint2d center(0.0, 0.0);
+  double radius_squared = 0.0;
+
+  auto cross = [](lanelet::BasicPoint2d p1, lanelet::BasicPoint2d p2) -> double {
```

Comment:
I think it's  better to use const
```suggestion
  auto cross = [](const lanelet::BasicPoint2d &p1, const lanelet::BasicPoint2d &p2) -> double {
```

Reply:
To apply this suggestion, is it need to change the `cross` parameters to have memory allocation?
```
const auto p4 = p2 - p1;
const auto p5 = p3 - p1;
const double S = cross(p4, p5);
```

Reply:
Or, I would appreciate it if you come up with another algorithm for calculating the center coordinate of the circle through three point.

Reply:
@veqcc 
I changed this suggestion to const